app/src/main/java/com/puzzlebench/cleanMarvelKotlin/data/db/RealmImpl.kt

```diff
@@ -0,0 +1,38 @@
+package com.puzzlebench.cleanMarvelKotlin.data.db
+
+import com.puzzlebench.cleanMarvelKotlin.data.db.entities.CharacterEntity
+import com.puzzlebench.cleanMarvelKotlin.data.mapper.CharacterMapperDatabase
+import com.puzzlebench.cleanMarvelKotlin.domain.model.Character
+import io.reactivex.Observable
+import io.realm.Realm
+import io.realm.kotlin.where
+
+class RealmImpl(private val mapper: CharacterMapperDatabase = CharacterMapperDatabase()) {
+
+    @Suppress("NAME_SHADOWING")
+    fun addCharacters(list: List<Character>): Observable<Boolean> {
+        return Observable.create { subscriber ->
+            val realmInstance = Realm.getDefaultInstance()
+            realmInstance.executeTransaction { realmInstance ->
+                realmInstance.deleteAll()
+            }
+            for (item in list) {
+                realmInstance.executeTransaction { realmInstance ->
+                    realmInstance.insertOrUpdate(mapper.transformToResponse(item))
+                }
+            }
+            subscriber.onNext(true)
```

Comment:
Please add a check to know if the DB was inserted fine

Reply:
Done!